packages/core/build.gradle.kts

```diff
@@ -11,162 +11,39 @@
  * License for the specific language governing permissions and limitations under the License.
  */
 
-@file:Suppress(
-  "UnstableApiUsage",
-  "unused",
-  "UNUSED_VARIABLE",
-  "DSL_SCOPE_VIOLATION",
-  "OPT_IN_USAGE",
-)
-
-import ElidePackages.elidePackage
+import elide.internal.conventions.elide
+import elide.internal.conventions.kotlin.*
 
 plugins {
+  id("elide.internal.conventions")
+
+  kotlin("multiplatform")
   kotlin("kapt")
-  id("dev.elide.build")
-  id("dev.elide.build.publishable")
-  id("dev.elide.build.multiplatform")
 }
 
-group = "dev.elide"
-version = rootProject.version as String
-
-val buildMingw = project.properties["buildMingw"] == "true"
-val buildWasm = project.properties["buildWasm"] == "true"
-val kotlinVersion: String = project.properties["versions.kotlin.sdk"] as String
-
-kotlin {
-  explicitApi()
-
-  jvm {
-    withJava()
-  }
-
-  js {
-    nodejs()
-    browser()
-    generateTypeScriptDefinitions()
-
-    compilations.all {
-      kotlinOptions {
-        sourceMap = true
-        moduleKind = "umd"
-        metaInfo = true
-      }
-    }
-  }
-
-  macosArm64()
-  iosArm64()
-  iosX64()
-  watchosArm32()
-  watchosArm64()
-  watchosX64()
-  tvosArm64()
-  tvosX64()
-  if (buildMingw) mingwX64()
-
-  if (buildWasm) {
-    wasmJs {
-      nodejs()
-      d8()
-      browser()
-    }
-    wasmWasi {
-      // nothing yet
-    }
+elide {
+  publishing {
+    id = "core"
+    name = "Elide Core"
+    description = "Pure Kotlin utilities provided across all supported platforms for the Elide Framework."
   }
 
-  sourceSets {
-    val commonMain by getting {
-      dependencies {
-        implementation(kotlin("stdlib"))
-      }
-    }
-    val commonTest by getting {
-      dependencies {
-        implementation(kotlin("stdlib"))
-        implementation(kotlin("test"))
-
-        configurations["kapt"].dependencies.add(
-          mn.micronaut.inject.java.asProvider().get()
-        )
-      }
-    }
-    val jvmMain by getting {
-      dependencies {
-        api(kotlin("stdlib-jdk8"))
-        api(libs.jetbrains.annotations)
-      }
-    }
-    val jvmTest by getting {
-      dependencies {
-        implementation(kotlin("stdlib-jdk8"))
-        implementation(kotlin("test"))
-      }
```

Comment:
two questions:

1) is all of this complexity legitimately handled in the new plugin? if so, great work
2) do we still get generic source set dependency warnings? i.e. depending on `nativeTest`

Reply:
1. yes and no, [here](https://github.com/elide-dev/elide/blob/61f90ad21e02aa56f8cc55a5d571f72a569e178e/tools/elide-build/src/main/kotlin/elide/internal/conventions/kotlin/KotlinMultiplatform.kt#L96) we automatically add the `stdlib` and `test` dependencies to the common source sets, which will cause those dependencies to be resolved into platform-specific artifacts on dependent source sets such as `jvmMain`. However, this particular project uses `stdlib-jdk8`, which I just noticed is _not_ handled by the conventions (and it should, will get that done).
2. as far as i can see at the moment, we only receive warnings from a deprecated feature being used by the KSP plugin. Do you mean the warnings that started with `1.9.20` about KMP source set dependencies?

Reply:
(btw we'll need to omit kotlin stdlib from maven artifacts before we ship, but i can cover that one)

Reply:
and yes those are the warnings i mean:

```
w: Kotlin Source Set 'tvosX64Main' can't depend on 'nativeMain' which is a default source set for compilation.
None of source sets can depend on the compilation default source sets.
Please remove this dependency edge.
```